### PR TITLE
Validate address prefix against xbridge config

### PR DESCRIPTION
**Pull request by magic53**
Compare the address prefix (version number) with the user specified address. This will prevent the situation where a user puts in the wrong coin address (or incompatible address).

### CODE REVIEW

File: src/xbridge/xbridgewalletconnector.cpp [hunk 1]
```diff
@@ -51,4 +52,22 @@ double WalletConnector::getWalletBalance() const
...
+    if (addr.empty())
+        return false;
+    std::string addrChk = DecodeBase58(addr.c_str());
+    std::string prefix(addrChk.begin(), addrChk.begin()+2);
+    stringstream o1; o1 << std::hex << prefix;
+    int ah; o1 >> ah;
+    int ac = (int)addrPrefix[0];
+    return ah == ac;
+}
+
```
Comment (corvaxx) [inline src/xbridge/xbridgewalletconnector.cpp]: OMG!!!

bool Connector::validateAddress(const std::string & address)
{
    std::vector<unsigned char> decoded;
    if (!DecodeBase58Check(address, decoded))
    {
        return false;
    }

    bool isP2PKH = memcmp(addrPrefix,   &decoded[0], decoded.size()-sizeof(uint160)) == 0;
    bool isP2SH  = memcmp(scriptPrefix, &decoded[0], decoded.size()-sizeof(uint160)) == 0;

    return isP2PKH || isP2SH;
}

Comment (corvaxx) [reply src/xbridge/xbridgewalletconnector.cpp]: second variant
```
bool Connector::validateAddress(const std::string & address)
{
    return rpc::validateaddress(m_user, m_passwd, m_ip, m_port, address);
}
```

Comment (magic53) [reply src/xbridge/xbridgewalletconnector.cpp]: `memcmp` + P2SH 👍 , can you commit over this branch? will put this in

Comment (magic53) [reply src/xbridge/xbridgewalletconnector.cpp]:
```bool WalletConnector::hasValidAddressPrefix(const std::string &addr) const {
    std::vector<unsigned char> decoded;
    if (!DecodeBase58Check(address, decoded))
    {
        return false;
    }

    bool isP2PKH = memcmp(addrPrefix,   &decoded[0], decoded.size()-sizeof(uint160)) == 0;
    bool isP2SH  = memcmp(scriptPrefix, &decoded[0], decoded.size()-sizeof(uint160)) == 0;

    return isP2PKH || isP2SH;
}```

Comment (corvaxx) [reply src/xbridge/xbridgewalletconnector.cpp]: sorry, i was not check this code. it's example

Comment (magic53) [reply src/xbridge/xbridgewalletconnector.cpp]: I checked, it works well. `memcmp` is faster. I can commit but want to give you credit

Comment (corvaxx) [reply src/xbridge/xbridgewalletconnector.cpp]: do it :)

**Pull request merged**